.github/workflows/publish.yml

```diff
@@ -27,25 +27,97 @@ jobs:
             echo "Error: tag does not match semver"
             exit 1
           fi
-  publish:
+
+  # If we use the viamrobotics/build-action@v1, it farms out compilation to Linux machines no
+  # matter which OS you're targeting (which is fine for Go and Python, but not for C++).
+  # Furthermore, these Linux machines do not cache build artifacts from one run to another. So
+  # instead, we're going to build everything on Github's action runners directly.
+  build-windows:
+    name: Build for Windows
     needs: validate-tag
-    name: Build and Upload module
-    runs-on: ubuntu-latest
+    runs-on: windows-2019
+    steps:
+    - uses: actions/checkout@v4
+    - name: Restore cached libraries
+      uses: actions/cache@v3
+      with:
+        path: C:\Users\runneradmin\.conan2
+        key: ${{ runner.os }}-windows2019-conan-${{ hashFiles('**/conanfile.py') }}
+        restore-keys: |
+          ${{ runner.os }}-windows2019-conan-
```

Comment:
Part of me wants to put a hyphen between `windows` and `2019` so it matches lines 38 and 58 (and 108). Part of me doesn't want to change the cache key because it'll take another 2 hours to build everything on a fresh cache. Worth the hassle? I can go either way.